cmd/tail.go

```diff
@@ -35,15 +35,41 @@ type line struct {
 
 type lineWriter struct {
 	name string
+	buf  bytes.Buffer
 	out  chan line
 }
 
-func (w lineWriter) Write(b []byte) (int, error) {
-	s := strings.TrimSuffix(string(b), "\n")
-	w.out <- line{Line: tail.Line{Text: s, Time: time.Now()}, source: w.name}
+func (w *lineWriter) Write(b []byte) (int, error) {
+	// Aggregate lines, then submit all completed lines to the output channel.
+	w.buf.Write(b)
+	i := bytes.LastIndexAny(w.buf.Bytes(), "\r\n")
+	if i == -1 {
+		// Always return that we wrote all the bytes.
+		return len(b), nil
+	}
+
+	text := w.buf.Next(i)
+	// Consume \r, \n, or \r\n. Note that the Buffer takes care of re-using space when we catch up.
+	crOrLf, err := w.buf.ReadByte()
+	if err != nil {
+		return len(b), err
+	}
+
+	// If the last character was LF, check whether the character before was CR and if so remove it.
```

Comment:
Why do we need to remove CR? Comment would help. Might also help to place this code closer to the "Consume \r" comment since this seems to be handling the `\r\n` case.

Reply:
Can't really put it closer, but I'll update the comment.